### PR TITLE
Sdk fixes

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fixes error in form actions (https://github.com/RasaHQ/rasa-sdk/pull/113#issuecomment-558971050)
- fixes https://github.com/RasaHQ/rasa-sdk/issues/121

**Notes**:
- I did fixes here cause Rasa should be robust to action server responses (`None` instead of `[]` is not a huge outlier). However, we should still check the `rasa-sdk` 

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: tests/core/test_actions.py [hunk 1]
```diff
@@ -216,6 +216,52 @@ def test_domain_action_instantiation():
...
+    )
+    with aioresponses() as mocked:
+        mocked.post("https://example.com/webhooks/actions", payload=response)
+
+        events = await remote_action.run(
+            default_channel, nlg, default_tracker, default_domain
+        )
+
+    assert events == [
+        BotUttered("what dou want to eat?", data={"buttons": [], "elements": []}),
```
Comment (erohmensing) [inline tests/core/test_actions.py]: Is this normal, or are the buttons and elements not getting cleaned correctly (i think everything else disappears if it is `None`)?

**Comment by wochinge** [comment]
This is ready from my side @erohmensing

**Comment by erohmensing** [comment]
Cool, I think it is ready from mine as well. Let's wait for tests to pass

File: rasa/core/actions/action.py [hunk 2]
```diff
@@ -406,28 +406,26 @@ def _validate_action_result(self, result):
...
-                del response["template"]
             else:
                 draft = {}
 
-            if "buttons" in response:
-                if "buttons" not in draft:
-                    draft["buttons"] = []
-                draft["buttons"].extend(response["buttons"])
-                del response["buttons"]
+            buttons = response.pop("buttons", []) or []
```
Comment (erohmensing) [inline rasa/core/actions/action.py]: are both `[]` necessary here?

Comment (wochinge) [reply rasa/core/actions/action.py]: yes

Comment (wochinge) [reply rasa/core/actions/action.py]: pop gives in an error in case the key is not there

**Comment by fanlu** [comment]
Cool!

**Pull request merged**